src/Form/FormAnalyzer.js

```diff
@@ -277,6 +277,45 @@ class FormAnalyzer {
         }
         return this
     }
+
+    /** @type {undefined|boolean} */
+    _isCCForm = undefined
+    /**
+     * Tries to infer if it's a credit card form
+     * @returns {boolean}
+     */
+    isCCForm () {
+        if (this._isCCForm !== undefined) return this._isCCForm
```

Comment:
This is the only change to this method, apart from being moved here. Just memoizing the result. When the form needs to be rescanned it follows the same flow as `isLogin` and `isSignup`. The analyzer is destroyed and recreated, so form data is kept up to date when needed and only when needed.

Reply:
very nice!